wandb/sdk/wandb_init.py

```diff
@@ -48,10 +48,10 @@
 if TYPE_CHECKING:
     from wandb.proto import wandb_internal_pb2 as pb
 
-logger: logging.Logger | None = None  # logger configured during wandb.init()
```

Comment:
why did you change the type here?

Reply:
it is set to an EarlyLogger in _WandbInit.setup() (mypy easily detects this possibility now)

Reply:
oh i see... i think we need to replace this EarlyLogger with just regular logger that saves the logs in buffer instead of file until a new run is created, this is probably the correct solution here, but for now the type is correct.